packages/tangle/ledgerstate.go

```diff
@@ -266,6 +269,26 @@ func (l *LedgerState) Consumers(outputID ledgerstate.OutputID) (cachedTransactio
 	return l.UTXODAG.CachedConsumers(outputID)
 }
 
+// ConfirmedConsumer returns the (cached) confirmed consumer of the given outputID.
+func (l *LedgerState) ConfirmedConsumer(outputID ledgerstate.OutputID) (confirmedConsumer *ledgerstate.CachedConsumer) {
+	found := false
+	for _, cachedConsumer := range l.Consumers(outputID) {
+		cachedConsumer.Consume(func(consumer *ledgerstate.Consumer) {
+			// We already identified a confirmed Consumer
+			l.CachedTransactionMetadata(consumer.TransactionID()).Consume(func(transactionMetadata *ledgerstate.TransactionMetadata) {
+				if l.tangle.ConfirmationOracle.IsBranchConfirmed(transactionMetadata.BranchID()) {
```

Comment:
I'm not sure if that always works. What happens if the transaction is in the master branch? It might be necessary to add another method `IsTransactionConfirmed` to the `ConfirmationOracle`.